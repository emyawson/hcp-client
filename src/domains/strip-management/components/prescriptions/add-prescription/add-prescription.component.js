import React from 'react';

import { colors } from 'src/core/styles/colors';
import { LocalizedText } from 'src/components';
import { stopPropagation } from 'src/utils';
import { PlusIcon } from 'src/assets/icons';

import {
  AddPrescriptionButton,
  CircleButton,
  CircleButtonSymbol,
} from './add-prescription.style';

export const AddButton = ({ clickHandler }) => (
  <CircleButton onClick={stopPropagation(clickHandler)}>
    <CircleButtonSymbol bold>
      <PlusIcon fillColor={colors.white} />
    </CircleButtonSymbol>
  </CircleButton>
);

export const AddPrescription = ({ disabled, expandHandler }) => (
  <AddPrescriptionButton
    role="button"
    onClick={expandHandler}
    disabled={disabled}
  >
    <AddButton clickHandler={expandHandler} />
    <span>
      <LocalizedText textKey="prescription.createNewTemporary" />
    </span>
  </AddPrescriptionButton>
);
