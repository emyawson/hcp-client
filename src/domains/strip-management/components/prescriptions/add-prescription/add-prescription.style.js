import styled from 'styled-components';

import {
  borderRadius,
  boxShadow,
  fontSize,
  colors,
  spacing,
  transitions,
} from 'src/core';
import { convertPxToRem, combineRems } from 'src/utils';

const colorActive = colors.brandBlue;
const colorDisabled = colors.grayLight;

export const CircleButton = styled.span`
  align-items: center;
  background-color: ${colorActive};
  border-radius: 50%;
  border: ${convertPxToRem(2)} solid ${colorActive};
  color: ${colorActive};
  display: inline-flex;
  justify-content: center;
  line-height: 1;
  outline: none;
  height: ${props => convertPxToRem(24)};
  width: ${props => convertPxToRem(24)};
  margin: 0 ${spacing.three} 0 0;
  cursor: pointer;
`;

export const CircleButtonSymbol = styled.span``;

export const AddPrescriptionButton = styled.button`
  align-items: center;
  background-color: ${colors.white};
  border: ${convertPxToRem(2)} solid
    ${props => (props.disabled ? colorDisabled : colorActive)};
  border-radius: ${borderRadius.three};
  cursor: pointer;
  display: flex;
  justify-content: center;
  font-size: ${fontSize.p};
  padding: ${combineRems(spacing.two, spacing.three)} ${spacing.three};
  transition: ${transitions.default};
  width: 100%;

  &:hover,
  &:focus {
    box-shadow: ${boxShadow({ depth: 'one' })};
    border-color: ${colorActive};
    color: ${colorActive};
    outline: none;
  }

  &[disabled] {
    color: ${colorDisabled};
    cursor: not-allowed;
    pointer-events: none;

    ${CircleButton} {
      background-color: ${colorDisabled};
      border-color: ${colorDisabled};
      color: ${colorDisabled};
    }
  }
`;
