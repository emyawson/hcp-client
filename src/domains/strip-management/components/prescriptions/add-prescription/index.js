export * from './add-prescription.component';
