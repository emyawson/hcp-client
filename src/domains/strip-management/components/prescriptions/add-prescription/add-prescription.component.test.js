import React from 'react';
import { shallow, mount } from 'enzyme';

import { AddPrescription } from './add-prescription.component';
import { CircleButton } from './add-prescription.style';

describe('Add prescription component', () => {
  it('Renders correctly', () => {
    const mockProps = {
      disabled: false,
      expandHandler: () => {},
    };
    const wrapper = shallow(<AddPrescription {...mockProps} />);
    expect(wrapper).toMatchSnapshot();
  });
  it('Clicking on component calls click handler', () => {
    const mockHandler = jest.fn().mockName('click handler');
    const mockProps = { disabled: false, expandHandler: mockHandler };
    const wrapper = shallow(<AddPrescription {...mockProps} />);
    wrapper.props().onClick();
    expect(mockHandler).toBeCalled();
  });
  it('Clicking on button within component calls click handler', () => {
    const mockHandler = jest.fn().mockName('another click handler');
    const mockProps = { disabled: false, expandHandler: mockHandler };
    const wrapper = mount(<AddPrescription {...mockProps} />);
    wrapper
      .find(CircleButton)
      .props()
      .onClick({
        stopPropagation: () => {},
        nativeEvent: { stopImmediatePropagation: () => {} },
      });
    expect(mockHandler).toBeCalled();
  });
});
