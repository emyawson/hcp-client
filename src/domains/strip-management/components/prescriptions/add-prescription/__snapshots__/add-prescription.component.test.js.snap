// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Add prescription component Renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    disabled={false}
    expandHandler={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <Unknown
          clickHandler={[Function]}
        />,
        <span>
          <LocalizedText
            textKey="prescription.createNewTemporary"
          />
        </span>,
      ],
      "disabled": false,
      "onClick": [Function],
      "role": "button",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": null,
        "nodeType": "function",
        "props": Object {
          "clickHandler": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "children": <LocalizedText
            textKey="prescription.createNewTemporary"
          />,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": null,
          "nodeType": "function",
          "props": Object {
            "textKey": "prescription.createNewTemporary",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "span",
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <Unknown
            clickHandler={[Function]}
          />,
          <span>
            <LocalizedText
              textKey="prescription.createNewTemporary"
            />
          </span>,
        ],
        "disabled": false,
        "onClick": [Function],
        "role": "button",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": null,
          "nodeType": "function",
          "props": Object {
            "clickHandler": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": <LocalizedText
              textKey="prescription.createNewTemporary"
            />,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": null,
            "nodeType": "function",
            "props": Object {
              "textKey": "prescription.createNewTemporary",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "span",
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
